clap/multiarg.zig

```diff
@@ -0,0 +1,166 @@
+const clap = @import("../clap.zig");
+const std = @import("std");
+
+const testing = std.testing;
+const heap = std.heap;
+const mem = std.mem;
+const debug = std.debug;
+
+pub fn MultiArgComptimeClap(comptime Id: type, comptime params: []const clap.Param(Id)) type {
```

Comment:
I've thought a little more about this. I think we just make `MultiArgComptimeClap` into the new `ComptimeClap`. If we wanna have `takes_value` as an enum then I don't see much reason why we need two parsers.

Reply:
Sorry for making you pull out your code and then make you put it back in 😅

Reply:
All part of the process! 😸